APIHook.cpp

```diff
@@ -664,34 +703,40 @@ class ChronosFileSaveDialog : public IFileSaveDialog
 			}
 
 			CString strSelPath(wstrSelPath);
-			CString strSelPathUpper(wstrSelPath);
-			CoTaskMemFree(wstrSelPath);
-			strSelPathUpper.MakeUpper();
-			if (strSelPathUpper.IsEmpty())
+			if (theApp.IsSGMode())
 			{
-				return hresult;
-			}
+				CString strSelPathUpper(wstrSelPath);
+				CoTaskMemFree(wstrSelPath);
+				strSelPathUpper.MakeUpper();
+				if (strSelPathUpper.IsEmpty())
+				{
+					return hresult;
+				}
+				CString strRoot(strRootPath);
+				strRoot.MakeUpper();
+				if (strSelPathUpper.Find(strRoot) != 0)
+				{
+					CString strMsg;
+					strMsg.Format(L"%sドライブ以外は指定できません。\n\n保存する場所から%sを指定しなおしてください。\n\n選択された場所[%s]", (LPCWSTR)strRootPath, (LPCWSTR)strRootPath, (LPCWSTR)strSelPath);
+					::MessageBoxW(hwndOwner, strMsg, theApp.m_strThisAppName, MB_OK | MB_ICONWARNING);
+					continue;
+				}
 
-			CString strRoot(strRootPath);
-			strRoot.MakeUpper();
-			if (strSelPathUpper.Find(strRoot) != 0)
-			{
-				CString strMsg;
-				strMsg.Format(L"%sドライブ以外は指定できません。\n\n保存する場所から%sを指定しなおしてください。\n\n選択された場所[%s]", (LPCWSTR)strRootPath, (LPCWSTR)strRootPath, (LPCWSTR)strSelPath);
-				::MessageBoxW(hwndOwner, strMsg, theApp.m_strThisAppName, MB_OK | MB_ICONWARNING);
-				continue;
+				CStringW strTSG_Upload = strRootPath + L"Upload\\";
+				CStringW strTSG_UploadUpper(strTSG_Upload);
+				strTSG_UploadUpper.MakeUpper();
+				if (strSelPathUpper.Find(strTSG_UploadUpper) == 0)
+				{
+					CString strMsg;
+					strMsg.Format(L"アップロードフォルダー[%s]には保存できません。\n\n指定しなおしてください。\n\n選択された場所[%s]", (LPCWSTR)strTSG_Upload, (LPCWSTR)strSelPath);
+					::MessageBoxW(hwndOwner, strMsg, theApp.m_strThisAppName, MB_OK | MB_ICONWARNING);
+					continue;
+				}
 			}
 
-			CStringW strTSG_Upload = strRootPath + L"Upload\\";
-			CStringW strTSG_UploadUpper(strTSG_Upload);
-			strTSG_UploadUpper.MakeUpper();
-			if (strSelPathUpper.Find(strTSG_UploadUpper) == 0)
-			{
-				CString strMsg;
-				strMsg.Format(L"アップロードフォルダー[%s]には保存できません。\n\n指定しなおしてください。\n\n選択された場所[%s]", (LPCWSTR)strTSG_Upload, (LPCWSTR)strSelPath);
-				::MessageBoxW(hwndOwner, strMsg, theApp.m_strThisAppName, MB_OK | MB_ICONWARNING);
-				continue;
-			}
+			PathRemoveFileSpec(strSelPath.GetBuffer());

```

Comment:
https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathremovefilespecw

> Note  This function is deprecated. We recommend the use of the [PathCchRemoveFileSpec](https://learn.microsoft.com/en-us/windows/desktop/api/pathcch/nf-pathcch-pathcchremovefilespec) function in its place.

PathRemoveFileSpec seems deprecated. Can we use alternative one?

Reply:
I think we can use alternative one.

However, `PathRemoveFileSpec` is used many places in Chronos.
I want to fix them in another PR.

Reply:
It is reasonable action. 
Le'ts do so.